resim/utils/tuple_utils_test.cc

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 ReSim, Inc.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+#include "resim/utils/tuple_utils.hh"
+
+#include <gtest/gtest.h>
+
+namespace resim {
+
+TEST(TupleUtilsTest, TestForEachInTuple) {
+  // SETUP
+  // NOLINTBEGIN(readability-magic-numbers)
+  auto simple_tuple = std::make_tuple(1, 2., 3U);
+  // NOLINTEND(readability-magic-numbers)
+  const auto add_one = [](auto number) { return number + 1; };
+
+  // ACTION
+  auto result = for_each_in_tuple(add_one, simple_tuple);
+
+  // VERIFICATION
+  static_assert(
+      std::tuple_size_v<decltype(simple_tuple)> ==
+      std::tuple_size_v<decltype(result)>);
+
+  EXPECT_EQ(std::get<0>(simple_tuple) + 1, std::get<0>(result));
```

Comment:
(nit) is it worth explictly stating and verifying the types?

Reply:
Sure thing. I've added a static assert for this!